client/engine/chainservice/simplechainservice.go

```diff
@@ -34,16 +31,9 @@ func (mcs simpleChainService) Out() <-chan Event {
 	return mcs.out
 }
 
-// In returns the in chan but narrows the type so that external consumers may only send on it.
-func (mcs simpleChainService) In() chan<- protocols.ChainTransaction {
-	return mcs.in
-}
-
-// forwardTransactions pipes transactions to the MockChain
-func (mcs simpleChainService) forwardTransactions() {
-	for tx := range mcs.in {
-		mcs.chain.In() <- tx
-	}
+// Send pipes transactions to the MockChain
+func (mcs simpleChainService) Send(tx protocols.ChainTransaction) {
+	mcs.chain.In() <- tx // TODO block until this has been successful / convert chain.In to a sync call
```

Comment:
I believe this does block until successful. 

https://go.dev/tour/concurrency/2:
> By default, sends and receives block until the other side is ready.

ie, this code does not advance until `In()` is ready to consume `tx`.

```suggestion
	mcs.chain.In() <- tx // TODO convert chain.In to a sync call
```